server/channels/app/plugin.go

```diff
@@ -438,6 +438,14 @@ func (ch *Channels) enablePlugin(id string) *model.AppError {
 
 	ch.cfgSvc.UpdateConfig(func(cfg *model.Config) {
 		cfg.PluginSettings.PluginStates[id] = &model.PluginState{Enable: true}
+
+		if _, ok := cfg.PluginSettings.Plugins[id]; !ok {
+			cfg.PluginSettings.Plugins[id] = make(map[string]any)
+		}
+
+		for _, pluginSetting := range manifest.SettingsSchema.Settings {
+			cfg.PluginSettings.Plugins[id][pluginSetting.Key] = pluginSetting.Default
+		}
 	})
```

Comment:
I wonder if we can make it so this is done on plugin installation, rather than plugin enable. This would allow a script to populate the plugin settings before enabling the plugin. What do you think @ChandanChainani?

Reply:
@mickmister Yes, I was thinking the same on start however nothing is done with config during installation so that's why instead I have gone with the enable part where plugin state in config is already being updated.